plugins/gdtf/gdtf-helpers.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @param {!object} startNode The XML object the reference should be resolved against.
+ * @param {!string} nodeReference A string of the form "Name.Name.Name…", see https://gdtf-share.com/wiki/GDTF_File_Description#attrType-node
+ * @returns {?object} The referenced XML node object, or null if it could not be found.
+ */
+function followXmlNodeReference(startNode, nodeReference) {
+  if (!startNode || !nodeReference) {
+    return null;
+  }
+
+  const nameParts = nodeReference.split(`.`);
+  let currentNode = startNode;
+
+  for (const nameAttr of nameParts) {
+    const nodeWithNameAttr = getChildNodes(currentNode).find(
+      node => `$` in node && node.$.Name === nameAttr
+    );
+
+    if (nodeWithNameAttr) {
+      currentNode = nodeWithNameAttr;
+    }
+    else {
+      return null;
+    }
+  }
+
+  return currentNode;
+
+
+  /**
+   * @param {!object} node The XML object.
+   * @returns {!array.<!object>} The XML objects of this node's child nodes.
+   */
+  function getChildNodes(node) {
+    return [].concat(
+      ...Object.keys(node).filter(
+        tagName => tagName !== `$`
+      ).map(
+        tagName => node[tagName]
+      )
+    );
+  }
+}
+
+
+/**
+ * Convert from CIE color representation xyY 1931 to RGB.
+ * See https://wolfcrow.com/blog/what-is-the-difference-between-cie-lab-cie-rgb-cie-xyy-and-cie-xyz/
+ * @param {!string} gdtfColorStr A string in the form "0.3127, 0.3290, 100.0", see https://gdtf-share.com/wiki/GDTF_File_Description#attrType-colorCIE
+ * @returns {!string} The RGB hex code string in the form "#rrggbb".
+ */
+function getRgbColorFromGdtfColor(gdtfColorStr) {
+  /* eslint-disable camelcase, space-in-parens */
+
+  // functions ported from https://github.com/njsmith/colorspacious
+  const sRGB1_to_sRGB255 = RGB1 => RGB1.map(c => c * 255);
+  const sRGB1_linear_to_sRGB1 = RGB_linear => RGB_linear.map(c => {
+    if (c <= 0.0031308) {
+      return 12.92 * c;
+    }
+
+    const a = 0.055;
+
+    return ((1 + a) * Math.pow(c, 1 / 2.4)) - a;
+  });
+  const XYZ100_to_sRGB1_linear = ([X, Y, Z]) => {
+    const R = ( 3.2406 * X / 100) + (-1.5372 * Y / 100) + (-0.4986 * Z / 100);
+    const G = (-0.9689 * X / 100) + ( 1.8758 * Y / 100) + ( 0.0415 * Z / 100);
+    const B = ( 0.0557 * X / 100) + (-0.2040 * Y / 100) + ( 1.0570 * Z / 100);
+    return [R, G, B];
+  };
+  const XYZ1_to_XYZ100 = XYZ1 => XYZ1.map(c => c * 100);
+  const xyY_to_XYZ = ([x, y, Y]) => {
```

Comment:
Since I ported the function from the [Python colorspacious package](https://github.com/njsmith/colorspacious/blob/master/colorspacious/basics.py#L98-L105), I'd rather keep the name the same to ease finding the original implementation again.